code/modules/research/xenobiology/crossbreeding/regenerative.dm

```diff
@@ -257,24 +408,52 @@ Regenerative extracts:
 	effect_desc = "Fully heals the target and also heals the user."
 
 /obj/item/slimecross/regenerative/lightpink/core_effect(mob/living/target, mob/user)
+	. = ..()
+	if (!.)
+		return FALSE
+
 	if(!isliving(user))
 		return
 	if(target == user)
 		return
-	var/mob/living/U = user
-	U.revive(HEAL_ALL)
-	to_chat(U, span_notice("Some of the milky goo sprays onto you, as well!"))
+	if (ishuman(user))
```

Comment:
Early return, but also, that first `!isliving(user)` check looks quite redundant now

But also this is making it so non-living mobs don't get self-healed by this, which is A change